civis/tests/mocks.py

```diff
@@ -37,6 +38,58 @@ def create_client_mock(cache=TEST_SPEC):
     return mock_client
 
 
+def create_client_container_mock(script_id=-10, run_id=100, state='succeeded',
+                                 run_outputs=None, log_outputs=None):
+    """Return a Mock set up for use in testing container scripts
+
+    Parameters
+    ----------
+    script_id: int
+        Mock-create containers with this ID when calling `post_containers`
+        or `post_containers_runs`.
+    run_id: int
+        Mock-create runs with this ID when calling `post_containers_runs`.
+    state: str, optional
+        The reported state of the container run
+    run_outputs: list, optional
+        List of Response objects returned as run outputs
+
+    Returns
+    -------
+    `unittest.mock.Mock`
+        With scripts endpoints `post_containers`, `post_containers_runs`,
+        `post_cancel`, and `get_containers_runs` set up.
+    """
+    c = mock.Mock()
+    c.__class__ = APIClient
```

Comment:
How about starting with the auto-specced client mock? The autospecs can be very helpful with testing. There's also a couple of places where the return of this function is now replacing `create_client_mock()`, so those tests would lose their autospecs.
```suggestion
    c = create_client_mock()
```